nsxt/resource_nsxt_ip_block_subnet.go

```diff
@@ -0,0 +1,172 @@
+	return &schema.Resource{
+		Create: resourceNsxtIpBlockSubnetCreate,
+		Read:   resourceNsxtIpBlockSubnetRead,
+		// Update ip block subnet is not supported by the NSX
+		Delete: resourceNsxtIpBlockSubnetDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+			},
+			"tag": getTagsSchemaForceNew(),
+			"block_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Block id for which the subnet is created",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"size": &schema.Schema{
+				Type:         schema.TypeInt,
+				Description:  "Represents the size or number of ip addresses in the subnet",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validatePowerOf2(),
+			},
+			"cidr": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Represents network address and the prefix length which will be associated with a layer-2 broadcast domain",
+				Computed:    true,
+			},
+			"allocation_ranges": &schema.Schema{
+				Type:        schema.TypeList,
+				Description: "A collection of IPv4 Pool Ranges",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "The start IP Address of the IP Range",
+							Computed:    true,
+						},
+						"end": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "The end IP Address of the IP Range",
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNsxtIpBlockSubnetCreate(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(*api.APIClient)
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	blockId := d.Get("block_id").(string)
+	size := int64(d.Get("size").(int))
+	tags := getTagsFromSchema(d)
+	ipBlockSubnet := manager.IpBlockSubnet{
+		DisplayName: displayName,
+		Description: description,
+		BlockId:     blockId,
+		Size:        size,
+		Tags:        tags,
+	}
+
+	ipBlockSubnet, resp, err := nsxClient.PoolManagementApi.CreateIpBlockSubnet(nsxClient.Context, ipBlockSubnet)
+
+	if err != nil {
+		return fmt.Errorf("Error during IpBlockSubnet create: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("Unexpected status returned during IpBlockSubnet create: %v", resp.StatusCode)
+	}
+	d.SetId(ipBlockSubnet.Id)
+
```

Comment:
I wonder if d.Set("block_id", blockId) here might work around the nsx bug (won't work for read though, not sure how bad this is)

Reply:
Since the read is always called after create/update I don't think it will help. But anyway we will not fully support IPAM in this NSX version, and it will be fixed for the next one.